### ISSUE TITLE
allow configuration of max peek size

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#4924]
This would allow instances to render bigger html files amongst other things.

Raised by Peter Briggs on [devlist](http://dev.list.galaxyproject.org/Enable-viewing-of-large-HTML-files-td4671161.html)

The value is now hardcoded here I believe: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/managers/hdas.py#L222

**Comment by sveinugu** [comment]
Code is moved slightly, now here, and still hardcoded: https://github.com/galaxyproject/galaxy/blob/ea0fe8053bd5772764ee13c120eeecab4c3574af/lib/galaxy/managers/hdas.py#L307

**Comment by mvdbeek** [comment]
I don't think we want to make this configurable, and I don't think this makes sense for HTML anyway, where you always need the whole document ?

**Comment by sveinugu** [comment]
> I don't think we want to make this configurable, and I don't think this makes sense for HTML anyway, where you always need the whole document ?

I'm not following you. The way I think about it is that is is exactly because you need the whole HTML document that you would like to make this configurable (or some similar functionality). In our case, we have a tool that creates a huge HTML report.

**Comment by mvdbeek** [comment]
Right, so you don't need to pass through the datatype-specific chunking.

**Comment by mvdbeek** [comment]
Maybe you could tell us what this is needed for exactly ? There's very limited use being made of `text_data`, its use should not affect `display_by_username_and_slug` for html files, why only leaves the markdown preview of textual content ... and it seems wrong to even include html that way

**Comment by mvdbeek** [comment]
or to say this in yet another way

> This would allow instances to render bigger html files amongst other things.

may have been true in 2017, but I don't think it is now ?

**Comment by sveinugu** [comment]
> may have been true in 2017, but I don't think it is now ?

This might very well be it. I didn't get the details in your previous comment. So the tool in question is this:

https://toolshed.g2.bx.psu.edu/view/sigven/oncoenrichr/023155e2e66c

It is installed in this Galaxy:

https://oncotools.elixir.no/

Which is running Galaxy 21.09. Previously (I can't remember if that was Galaxy 21.09 or a previous version), we needed to adjust the `MAX_PEEK_SIZE` to get the tool to work, and I just got a request on how to do this due to an error. But when I look at this more closely now, it seems the current issue is actually a full disk.

Sorry for the confusion. I guess we then can close down this issue when we get the tool to work without changing this.

**Comment by sveinugu** [comment]
So this is still an issue, at least in version `21.09` (and no indication from the code that this is different in `dev`). However, it was the wrong hard-coded value. This is the value that we need to change to allow HTML outputs larger than 10mb:

https://github.com/galaxyproject/galaxy/blame/7404ed0578fd027ab63b6ed2fccd01e2230ee821/lib/galaxy/datatypes/data.py#L174

**Comment by mvdbeek** [comment]
That makes a lot more sense, but raising the limit is not the right thing. I think html files can be served by just opening and returning the file handle.

**Comment by dannon** [comment]
@sveinugu Check out https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/data.py#L173, looks like this would do it?  I think we might be able to close this.

**Comment by sveinugu** [comment]
> @sveinugu Check out https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/data.py#L173, looks like this would do it? I think we might be able to close this.

Well that was the line above my link, but no the max value is still hard coded to 10MB (and has been for a while, I just got confused by the initial comment in this issue). Our tool generates even larger HTML files (those R folks, go figure!)

**Comment by dannon** [comment]
@sveinugu Ahh!  Sorry, I missed the link -- I think Marius is right; there's no point in cutting off an html file stream midway, might as well just ship it.
